### PR TITLE
Match solc diagnostic levels

### PR DESCRIPTION
**Pull request by seanyoung**
The following errors are warnings in solc

 - unreachable code should be warning
 - function types with parameter/return names should be warning
 - unknown assembly flags should be warnings

### CODE REVIEW

File: src/sema/namespace.rs [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ impl Namespace {
                         .into_iter()
                         .map(|p| {
                             if let Some(name) = p.id {
-                                diagnostics.push(Diagnostic::error(
+                                diagnostics.push(Diagnostic::warning(
```
Comment (LucasSte) [inline src/sema/namespace.rs:1137]: Should success still be false here even if this is not an error anymore?

File: src/sema/namespace.rs [hunk 2]
```diff
@@ -1148,7 +1148,7 @@ impl Namespace {
...
```
Comment (LucasSte) [inline src/sema/namespace.rs:1151]: Should success still be false here even if this is not an error anymore?

File: src/sema/namespace.rs (hunk 1 above)
Comment (seanyoung) [reply src/sema/namespace.rs:1137]: Good point, fixed

File: src/sema/namespace.rs (hunk 2 above)
Comment (seanyoung) [reply src/sema/namespace.rs:1150]: Good point, fixed

**Review by LucasSte — APPROVED** [review]

**Review by xermicus — APPROVED** [review]

**Pull request merged**